core/src/main/java/org/apache/iceberg/Partitioning.java

```diff
@@ -200,6 +200,16 @@ public Void alwaysNull(int fieldId, String sourceName, int sourceId) {
   /**
    * Builds a grouping key type considering all provided specs.
    *
+   * @param specs one or many specs
+   * @return the constructed grouping key type
+   */
+  public static StructType groupingKeyType(Collection<PartitionSpec> specs) {
```

Comment:
Do we consider this deprecated? When would we want to use it without passing Schema?

Reply:
We could, I wasn't sure. If you have a set of specs, we probably have a schema too. I'll deprecate.

Reply:
Deprecated.